cdap-ui/app/cdap/components/Reports/ReportsDetail/Runs/index.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {GLOBALS} from 'services/global-constants';
+import {humanReadableDate} from 'services/helpers';
+
+require('./Runs.scss');
+
+const PIPELINES = [GLOBALS.etlDataPipeline, GLOBALS.etlDataStreams];
+
+function getName(run) {
+  let name = run.application.name;
+
+  if (PIPELINES.indexOf(run.artifact.name) == -1) {
+    name = `${name} - ${run.program}`;
+  }
+
+  return name;
+}
+
+function getType(run) {
+  switch (run.artifact.name) {
+    case GLOBALS.etlDataPipeline:
+      return 'Batch Pipeline';
+    case GLOBALS.etlDataStreams:
+      return 'Realtime Pipeline';
+    default:
+      return run.type;
+  }
+}
+
+function RunsView({runs}) {
+  // this table will be having dynamic number of columns, so cannot use css grid
```

Comment:
nit: So right now if we don't provide an exact `grid-row` `grid-template-columns` it defaults to `1fr` for each column.

This might prevent horizontal scrolling as it will try to fit in all the columns within the viewport. @elfenheart was the grid layout failed on this or did you see any other issue?

Reply:
My understanding was that with the 1fr, all the columns will have the same width? That's why I didn't use the grid and just use table since table will attempt to size the column appropriately